lib/galaxy/app.py

```diff
@@ -581,8 +586,30 @@ def __init__(self, configure_logging=True, use_converters=True, use_display_appl
 
         self._configure_tool_shed_registry()
         self._register_singleton(tool_shed_registry.Registry, self.tool_shed_registry)
-        # Tool Data Tables
+        #  Tool Data Tables
         self._configure_tool_data_tables(from_shed_config=False)
+        self._register_celery_galaxy_task_components()
+
+    def _register_celery_galaxy_task_components(self):
+        """
+        Register subtype class instance to support implementation of a user rate limit for execution of celery tasks.
+        The default supertype class does not enforce a user rate limit. This is the case if the celery_user_rate_limit
+        config param is the default value.
+        """
+        if self.config.celery_user_rate_limit:
+            task_before_start: GalaxyTaskBeforeStart
+            if (
```

Comment:
We have a utility method `is_postgres` that does exactly that, but is located in the [wrong place](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/model/unittest_utils/model_testing_utils.py#L224). My suggestion is to move that method into `galaxy.model.database_utils` (there's only [one other place](https://github.com/galaxyproject/galaxy/blob/dev/test/unit/app/test_migrate_database.py#L11) where it's used - it would need to be adjusted), and use it here. Given that you wouldn't have the `celery_user_rate_limit_standard_before_start` config option, I think this code would become even simpler:
```py
if is_postgres(self.config.database_connection):
    task_before_start = GalaxyTaskBeforeStartUserRateLimitPostgres(
        self.config.celery_user_rate_limit, self.model.session
    )
else:
    task_before_start = GalaxyTaskBeforeStartUserRateLimitStandard(
        self.config.celery_user_rate_limit, self.model.session
    )
```

Reply:
I will make changes to use is_postgres. I would like to keep the celery_user_rate_limit_standard_before_start option. However, I guess I could never really see a need for it if we are using postgres as a database.